webhooks/tasks.py

```diff
@@ -0,0 +1,71 @@
+"""
+Celery tasks.
+"""
+import hashlib
+import hmac
+import json
+import logging
+
+from django.apps import apps
+from django.utils.encoding import force_bytes
+import requests
+
+from ccxcon.celery import async
+from webhooks.models import Webhook
+
+log = logging.getLogger(__name__)
+
+
+@async.task
+def publish_webhook(model_str, pk):
+    """
+    Publishes model updates to various services as defined by Webhook instances.
+    """
+    try:
+        app, model_label = model_str.split('.')
+    except ValueError:
+        log.warning("Malformed model string for webhook serialization: %s",
+                    model_str)
+        return
+
+    try:
+        model = apps.get_model(app, model_label)
+    except LookupError:
+        log.warning("Could not find the model %s for webhook publishing.",
+                    model_str)
+        return
+
+    try:
+        instance = model.objects.get(pk=pk)
+        if not hasattr(instance, 'to_webhook'):
+            log.info("Not submitting %s:%s to webhook. It does not support "
+                     "serialization", type(instance), pk)
+            return
+
+        payload = {
+            'action': 'update',
+            'type': model_label,
+            'payload': instance.to_webhook()
+        }
+
+    except model.DoesNotExist:
+        # Send delete.
+        payload = {
+            'action': 'delete',
+            'type': model_label,
+            'payload': {
+                'pk': pk
+            },
+        }
+    except model.MultipleObjectsReturned:
+        log.warning("Webhook lookup by PK unsuccessful. Returned multiple on pk "
+                    "lookup. <%s pk=%s>", model_str, pk)
+        return
+
+    for wh in Webhook.active.all():
+        j_payload = json.dumps(payload)
+        signature = hmac.new(force_bytes(wh.secret), force_bytes(j_payload),
+                             hashlib.sha1).hexdigest()
+        requests.post(wh.url, j_payload, headers={
```

Comment:
Also, you should use the `json` argument here instead of the `data` argument (which is where `j_payload` is going) to allow JSON with multiple layers of dicts. The `data` argument treats the input as form data which is probably not what we want